src/Microsoft.DotNet.Cli.Utils/CommandResolution/DepsJsonCommandResolver.cs

```diff
@@ -0,0 +1,261 @@
+            {
+                var runtimeAssetGroups = runtimeLibrary.RuntimeAssemblyGroups;
+
+                commandCandidates.AddRange(GetCommandCandidatesFromRuntimeAssetGroups(
+                    commandName, 
+                    runtimeAssetGroups,
+                    runtimeLibrary.Name,
+                    runtimeLibrary.Version));
+            }
+
+            return commandCandidates;
+        }
+
+        private IEnumerable<CommandCandidate> GetNativeCommandCandidates(string commandName, DependencyContext dependencyContext)
+        {
+            var commandCandidates = new List<CommandCandidate>();
+
+            foreach (var runtimeLibrary in dependencyContext.RuntimeLibraries)
+            {
+                var runtimeAssetGroups = runtimeLibrary.NativeLibraryGroups;
+
+                commandCandidates.AddRange(GetCommandCandidatesFromRuntimeAssetGroups(
+                                    commandName,
+                                    runtimeAssetGroups,
+                                    runtimeLibrary.Name,
+                                    runtimeLibrary.Version));            
+            }
+
+            return commandCandidates;
+        }
+
+        private IEnumerable<CommandCandidate> GetCommandCandidatesFromRuntimeAssetGroups(
+            string commandName,
+            IEnumerable<RuntimeAssetGroup> runtimeAssetGroups, 
+            string PackageName,
+            string PackageVersion)
+        {
+            var candidateAssetGroups = runtimeAssetGroups
+                .Where(r => r.Runtime == string.Empty)
+                .Where(a => 
+                    a.AssetPaths.Any(p => Path.GetFileNameWithoutExtension(p)
+                    .Equals(commandName, StringComparison.OrdinalIgnoreCase)));
+
+            var commandCandidates = new List<CommandCandidate>();
+            foreach (var candidateAssetGroup in candidateAssetGroups)
+            {
+                var candidateAssetPaths = candidateAssetGroup.AssetPaths.Where(
+                    p => Path.GetFileNameWithoutExtension(p)
+                    .Equals(commandName, StringComparison.OrdinalIgnoreCase));
+
+                foreach (var candidateAssetPath in candidateAssetPaths)
+                {
+                    commandCandidates.Add(new CommandCandidate
+                    {
+                        PackageName = PackageName,
+                        PackageVersion = PackageVersion,
+                        RelativeCommandPath = candidateAssetPath
+                    });
+                }
+            }
+
+            return commandCandidates;
+        }
+
+        private CommandCandidate ChooseCommandCandidate(IEnumerable<CommandCandidate> commandCandidates)
+        {
+            foreach (var extension in s_extensionPreferenceOrder)
+            {
+                var candidate = commandCandidates
+                    .FirstOrDefault(p => Path.GetExtension(p.RelativeCommandPath)
```

Comment:
Could you break the lines differently? Maybe move Path.GetExtensions... to the same line as Equals? I had to spend a couple seconds matching parenthesis.